src/toil/common.py

```diff
@@ -946,20 +943,20 @@ def _setBatchSystemEnvVars(self):
 
     def _serialiseEnv(self):
         """
-        Puts the environment in a globally accessible pickle file.
+        Puts the environment in a globally accessible json file.
         """
-        # Dump out the environment of this process in the environment pickle file.
-        with self._jobStore.writeSharedFileStream("environment.pickle") as fileHandle:
-            pickle.dump(os.environ, fileHandle, pickle.HIGHEST_PROTOCOL)
-        logger.info("Written the environment for the jobs to the environment file")
+        # Dump out the environment of this process in the environment json file.
+        with self._jobStore.writeSharedFileStream("environment.json") as fileHandle:
+            json.dump(dict(os.environ), fileHandle)
+        logger.debug("Written the environment for the jobs to the environment file")
```

Comment:
@DailyDreaming, This seems like a major enough change to have its own issue.

Reply:
Alright, I'll revert this segment.

Reply:
Should be reverted now.